tempesta_fw/http_msg.c

```diff
@@ -42,18 +42,39 @@ tfw_http_msg_make_hdr(TfwPool *pool, const char *name, const char *val)
 {
 	size_t n_len = strlen(name);
 	size_t v_len = val ? strlen(val) : 0;
-	const TfwStr tmp_hdr = {
-		.chunks = (TfwStr []){
-			{ .data = (void *)name,	.len = n_len },
-			{ .data = (void *)val,	.len = v_len,
-			  .flags = TFW_STR_HDR_VALUE},
-		},
-		.len = n_len + v_len,
-		.eolen = 2,
-		.nchunks = (val ? 2 : 1)
-	};
+	size_t n;
+	TfwStr *hdr, *h_c;
+	char *data;
+
+	n = ((val ? 2 : 1) + 1) * sizeof(TfwStr) + n_len + v_len;
+	hdr = (TfwStr *)tfw_pool_alloc(pool, n);
+	if (!hdr)
+		return NULL;
+	TFW_STR_INIT(hdr);
 
-	return tfw_strdup(pool, &tmp_hdr);
+	hdr->len = n_len + v_len;
+	hdr->eolen = 2;
+	hdr->nchunks = (val ? 2 : 1);
+	hdr->chunks = hdr + 1;
+	data = (char *)(TFW_STR_LAST(hdr) + 1);
+
+	h_c = TFW_STR_CHUNK(hdr, 0);
+
+	TFW_STR_INIT(h_c);
+	h_c->data = data;
+	h_c->len = n_len;
+	tfw_cstrtolower_wo_avx2(data, name, n_len);
+
+	data += h_c->len;
+	++h_c;
+
+	TFW_STR_INIT(h_c);
+	h_c->data = data;
+	h_c->len = v_len;
+	h_c->flags = TFW_STR_HDR_VALUE;
+	memcpy(data, val, v_len);
```

Comment:
If the code works in softirq, then we should avoid using `memcpy()` and similar functions: they're quite slow in the Linux kernel (much slower than in the user space with GCC/glibc), so if possible use our versions of the functions.

Reply:
This func call from process context wo `fpu_begin()`/`fpu_end()`